packages/dts-test/setupHelpers.test-d.ts

```diff
@@ -19,15 +19,15 @@ describe('defineProps w/ type declaration', () => {
   const props = defineProps<{
     foo: string
     bool?: boolean
-    boolAndUndefined: boolean | undefined
+    boolOrUndefined: boolean | undefined
   }>()
   // explicitly declared type should be refined
   expectType<string>(props.foo)
   // @ts-expect-error
   props.bar
 
   expectType<boolean>(props.bool)
-  expectType<boolean>(props.boolAndUndefined)
+  expectType<boolean>(props.boolOrUndefined)
```

Comment:
If the type is boolean or undefined then this line should be:
expectType<boolean | undefined >(props.boolOrUndefined)

Reply:
No. The name means the provided type is `boolean | undefined`, not finalized props type.